cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java

```diff
@@ -1304,6 +1308,222 @@ private void testSecureStorePipeline(Engine engine, String prefix) throws Except
     Assert.assertNotNull(getDataset(prefix + "MockSecureSinkDataset").get());
   }
 
+  @Test
+  public void testExternalDatasetTrackingMR() throws Exception {
```

Comment:
Shouldn't we include `ETLWorkflowTestRun#testInvalidTransformConfigFailsToDeploy` test?